integrations/oci/objectstorage/src/main/java/io/helidon/integrations/oci/objectstorage/OciObjectStorageHealthCheck.java

```
@@ -0,0 +1,233 @@
+        HealthCheckResponseBuilder builder = HealthCheckResponse.named("objectStorage");
+        if (bucket != null) {
+            // Attempt to retrieve bucket's metadata
+            Single<ApiOptionalResponse<GetObjectRx.Response>> single =
+                    ociObjectStorage.getBucket(GetObject.Request.builder().bucket(bucket));
+            try {
+                ApiOptionalResponse<GetObjectRx.Response> r = single.get(timeout, TimeUnit.SECONDS);
+                builder.state(r.status().equals(OK_200));
+                LOGGER.fine(() -> "OCI ObjectStorage health check for bucket " + bucket
+                        + " returned status code " + r.status().code());
+            } catch (Throwable t) {
+                builder.state(false);
+            } finally {
+                builder.withData("bucket", bucket);
+            }
+        } else {
+            LOGGER.fine("OCI ObjectStorage health check disabled when bucket not defined");
+            builder.state(true);
+        }
+        builder.withData("namespace", namespace);
+        return builder.build();
+    }
+
+    /**
+     * Fluent API builder for {@link OciObjectStorageHealthCheck}.
+     */
+    public static final class Builder implements io.helidon.common.Builder<OciObjectStorageHealthCheck> {
+
+        private Config config;
+        private String bucket;
+        private String namespace;
+        private int timeout = -1;
+        private OciObjectStorageRx ociObjectStorage;
+
+        private Builder() {
+        }
+
+        @Override
+        public OciObjectStorageHealthCheck build() {
+            return new OciObjectStorageHealthCheck(this);
+        }
+
+        /**
+         * Set up this builder using config.
+         *
+         * @param config the config.
+         * @return the builder.
+         */
+        public Builder config(Config config) {
+            this.config = config;
+            Config timeout = config.get("objectstorage.healthcheck.timeout");
```

config key for durations should contain the unit - such as
`timeout-millis` `timeout-seconds` so it is clear when setting it in a configuration file.

---

bucket, namespace etc. should be read here. The builder should throw an exception if either is undefined.